### PR TITLE
[cudaaligner] Add aligner.reset_bandwidth(int32_t) to banded Myers aligner

### PR DESCRIPTION
**Pull request by ahehn-nv**
* Make `batched_device_matrices` default-constructible.
* Add `aligner.reset_bandwidth(int32_t)` to banded Myers aligner.

Proper tests of `reset_bandwidth()` will follow in a separate PR.

### CODE REVIEW

**Comment by armintoepfer** [comment]
🚀

File: cudaaligner/src/aligner_global_myers_banded.cpp [hunk 1]
```diff
@@ -354,6 +375,19 @@ void AlignerGlobalMyersBanded::reset()
     alignments_.clear();
 }
 
+void AlignerGlobalMyersBanded::reset_bandwidth(const int32_t max_bandwidth)
+{
+    assert(max_device_memory_ >= 0);
```
Comment (mimaric) [inline cudaaligner/src/aligner_global_myers_banded.cpp:380]: What happens if `max_device_memory_ == 0` or small? Does the algorithm take care of that?

Comment (ahehn-nv) [reply cudaaligner/src/aligner_global_myers_banded.cpp:380]: max_device_memory is set by the constructor, which ensures that it `max_device_memory_ >= 0`.
If `max_device_memory_ == 0`, `add_alignment` will always fail and return the error `StatusType::exceeded_max_alignments`.

**Comment by ahehn-nv** [comment]
I added a commit to restrict the scope of the in-source warning disablers and added a new one to fix my PR. Please (re-)review.

File: cudaaligner/src/aligner_global_myers_banded.cpp (hunk 1 above)
Comment (mimaric) [reply cudaaligner/src/aligner_global_myers_banded.cpp:380]: Good, so that case is taken care of.

**Review by mimaric — APPROVED** [review]

**Comment by ahehn-nv** [comment]
rerun tests

**Comment by armintoepfer** [comment]
Can we propagate this function up to `claraparabricks::genomeworks::cudaaligner::Aligner`?

File: cudaaligner/src/aligner_global_myers_banded.cpp [hunk 2]
```diff
@@ -136,26 +121,62 @@ struct AlignerGlobalMyersBanded::InternalData
...
+    data->query_patterns = batched_device_matrices<WordType>(0, allocator, stream);
+
+    data->seq_d.clear_and_resize(2 * mem.sequence_memory / sizeof(char));
+    data->seq_starts_d.clear_and_resize(2 * n_alignments_initial + 1);
+    data->scheduling_index_d.clear_and_resize(n_alignments_initial);
+    data->scheduling_atomic_d.clear_and_resize(1);
+    data->results_d.clear_and_resize(mem.results_memory / sizeof(char));
+    data->result_starts_d.clear_and_resize(n_alignments_initial + 1);
+    data->result_lengths_d.clear_and_resize(n_alignments_initial);
+    data->pvs            = batched_device_matrices<WordType>(mem.pmvs_matrix_memory / sizeof(WordType), allocator, stream);
```
Comment (armintoepfer) [inline cudaaligner/src/aligner_global_myers_banded.cpp:163]: For whatever reason, when I call `reset_bandwidth` manually, it hangs here. When used via the ctor, it works fine.

**Pull request merged**